lib/netbox_client_ruby/communication.rb

```diff
@@ -45,8 +45,8 @@ def raise_on_http_error(response)
       body = response.body
 
       case status
-      when 200..399 then
-      when 400..499 then
+      when 200..299 then
+      when 300..499 then
```

Comment:
Why's this better? The original logic was that `200..399` are all considered success codes, whereas `400..499` are definitely failure codes.

Reply:
See https://redmine.nine.ch/issues/38280 for the issue.
We do not want to follow 302 redirects and as such we will treat them as errors.

Reply:
Of course. I miss-interpreted the ruby `case ... when` syntax. (It does not 'fall through' if nothing is in a `when` block.)

Reply:
I also assumed it would fall through when I first looked at it

Reply:
Apparently I can't alter that commit message as it refuses to push if i try to amend the last commit.